lib/elixir/lib/range.ex

```diff
@@ -461,6 +461,99 @@ defmodule Range do
     end
   end
 
+  @doc """
+  Checks if all members of `range2` are inside `range1`.
+
+  It returns true if all of the following conditions are met:
+
+   * the bounds of `range2` are fully within `range1`
+   * all members of `range2` also exist in `range1`
+
+
+      iex> Range.contains_members?(1..10, 2..5)
+      true
+
+      # the next 2 examples return false because the second range is not within
+      # the bounds of the first one
+      iex> Range.contains_members?(1..10, 11..12)
+      false
+      iex> Range.contains_members?(8..20, 1..12)
+      false
+
+  Steps are also considered when checking if the second range is contained by the
+  first:
+
+      iex> Range.contains_members?(1..10, 2..4//2)
+      true
+      iex> Range.contains_members?(2..8//2, 4..8//4)
+      true
+      # false returned because the second range doesn't start (3) at a point
+      # contained by the first (2, 4, 6 and 8)
+      iex> Range.contains_members?(2..8//2, 3..7//4)
+      false
+      iex> Range.contains_members?(2..11//3, 5..11//6)
+      true
+      iex> Range.contains_members?(11..2//-3, 5..11//6)
+      true
+      iex> Range.contains_members?(1..10//2, 2..8//3)
+      false
+      iex> Range.contains_members?(0..10//4, 0..4//2)
+      false
+  """
+  @spec contains_members?(t, t) :: boolean
+  def contains_members?(first1..last1//step1 = range1, first2..last2//step2 = range2) do
```

Comment:
`contains_members?` or `contains_points?`? In the rest of the Elixir standard library we call the elements in a collection "members".